tf2onnx/graph.py

```diff
@@ -1151,11 +1190,16 @@ def remove_input(node, to_be_removed, input_index=None):
         assert isinstance(node, Node) and isinstance(to_be_removed, six.text_type)
         if input_index is not None:
             assert node.input[input_index] == to_be_removed
+            if node.input[input_index] in self._input_to_node_name:
+                to_ops = self._input_to_node_name[node.input[input_index]]
+                if node.name in to_ops:
+                    to_ops.remove(node.name)
             del node.input[input_index]
             return True
 
         for i, name in enumerate(node.input):
             if name == to_be_removed:
+                self._unregister_input_name(node.input[i], node)
                 del node.input[i]
```

Comment:
lol I think this is a bug and will crash if a node has two identical inputs.  node.input should be traversed in reverse if using del.  Again not your code but might be worth fixing.

Reply:
I know this could be a bug. The current implementation does not do anything specific for that case, the unit test are not failing, I assumed this could be part of later refactoring. I did not want to mix too many modifications.

Reply:
I added an exception raised when this case happens.

Reply:
Awesome!